resources/functionMap.php

```diff
@@ -5953,41 +5953,41 @@
 'linkinfo' => ['int|false', 'filename'=>'string'],
 'litespeed_request_headers' => ['array'],
 'litespeed_response_headers' => ['array|false'],
-'Locale::acceptFromHttp' => ['string|false', 'header'=>'string'],
-'Locale::canonicalize' => ['string', 'locale'=>'string'],
-'Locale::composeLocale' => ['string', 'subtags'=>'array'],
-'Locale::filterMatches' => ['bool', 'langtag'=>'string', 'locale'=>'string', 'canonicalize='=>'bool'],
-'Locale::getAllVariants' => ['array', 'locale'=>'string'],
-'Locale::getDefault' => ['string'],
-'Locale::getDisplayLanguage' => ['string', 'locale'=>'string', 'in_locale='=>'string'],
-'Locale::getDisplayName' => ['string', 'locale'=>'string', 'in_locale='=>'string'],
-'Locale::getDisplayRegion' => ['string', 'locale'=>'string', 'in_locale='=>'string'],
+'Locale::acceptFromHttp' => ['non-empty-string|false', 'header'=>'string'],
+'Locale::canonicalize' => ['non-empty-string|null', 'locale'=>'string'],
+'Locale::composeLocale' => ['string|false', 'subtags'=>'array{language:string, script?:string, region?:string, variant?:array<array-key,string>, private?:array<array-key,string>, extlang?:array<array-key,string>, variant0?:string, variant1?:string, variant2?:string, variant3?:string, variant4?:string, variant5?:string, variant6?:string, variant7?:string, variant8?:string, variant9?:string, variant10?:string, variant11?:string, variant12?:string, variant13?:string, variant14?:string,  private0?:string, private1?:string, private2?:string, private3?:string, private4?:string, private5?:string, private6?:string, private7?:string, private8?:string, private9?:string, private10?:string, private11?:string, private12?:string, private13?:string, private14?:string, extlang0?:string, extlang1?:string, extlang2?:string}'],
```

Comment:
Narrowing down parameter types is a job for this file instead https://github.com/phpstan/phpstan-src/blob/1.11.x/resources/functionMap_bleedingEdge.php

Because it's essentially a BC break. You're forcing someone that calls this with a general `array` to update typehints in their whole call chain.

Reply:
In my tests, passing more parameters than the documented ones didn't reported anything, but I can move that to the bleeding edge 😄